### PR TITLE
feat(Redis): add new subnet_group_name input

### PR DESCRIPTION
**Pull request by AlexPiko**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -2,7 +2,7 @@ resource "aws_elasticache_replication_group" "redis" {
   engine = var.global_replication_group_id == null ? "redis" : null
 
   parameter_group_name = var.global_replication_group_id == null ? aws_elasticache_parameter_group.redis.name : null
-  subnet_group_name    = aws_elasticache_subnet_group.redis.name
+  subnet_group_name    = aws_elasticache_subnet_group.redis[0].name
```
Comment (g4rb1) [inline main.tf:5]:
```suggestion
  subnet_group_name    = try(var.subnet_group_name, aws_elasticache_subnet_group.redis[0].name)
```
https://developer.hashicorp.com/terraform/language/functions/try

**Review by g4rb1 — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (AlexPiko) [reply main.tf]: [9b9558b](https://github.com/StuDocu/terraform-aws-elasticache-redis/pull/5/commits/9b9558b5f24bd89a088a2ecdf632d7e1eb9d8de9)

File: main.tf [hunk 2]
```diff
@@ -2,7 +2,7 @@ resource "aws_elasticache_replication_group" "redis" {
   engine = var.global_replication_group_id == null ? "redis" : null
 
   parameter_group_name = var.global_replication_group_id == null ? aws_elasticache_parameter_group.redis.name : null
-  subnet_group_name    = aws_elasticache_subnet_group.redis.name
+  subnet_group_name    = var.subnet_group_name == null ? try(var.subnet_group_name, aws_elasticache_subnet_group.redis[0].name) : var.subnet_group_name
```
Comment (g4rb1) [inline main.tf:5]: the `try` function replaces the need for a conditional, null variables will be ignored
```suggestion
  subnet_group_name    = try(var.subnet_group_name, aws_elasticache_subnet_group.redis[0].name)
```

**Review by g4rb1 — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
Comment (AlexPiko) [reply main.tf]: [7ff1e12](https://github.com/StuDocu/terraform-aws-elasticache-redis/pull/5/commits/7ff1e12b0421307931ea7b6b4166a7ebc4df4e97)

**Review by g4rb1 — APPROVED** [review]

**Pull request merged**